src/lib/support/StorageKeyName.h

```diff
@@ -0,0 +1,78 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <lib/support/CHIPMemString.h>
+#include <stdio.h>
+
+namespace chip {
+
+/**
+ * Represents a key used for addressing a specific storage element.
+ *
+ * May contain generic fixed keys (e.g. "g/fidx") or formatted fabric-specific
+ * keys ("f/%x/..." where %x is the fabric index).
+ */
+class StorageKeyName
```

Comment:
Please bring this back in DefaultStorageKeyAllocator.

Reply:
done, thanks